nhsn/delphi_nhsn/pull.py

```diff
@@ -38,17 +47,31 @@ def check_last_updated(socrata_token, dataset_id, logger):
         client = Socrata("data.cdc.gov", socrata_token)
         response = client.get_metadata(dataset_id)
 
-        updated_timestamp = datetime.utcfromtimestamp(int(response["rowsUpdatedAt"]))
-        now = datetime.utcnow()
-        recently_updated_source = (now - updated_timestamp) < timedelta(days=1)
+        updated_timestamp = datetime.fromtimestamp(int(response["rowsUpdatedAt"]), tz=timezone.utc)
+
+        # pulling last updated from the api
+        meta_df = pd.DataFrame(Epidata.covidcast_meta()["epidata"])
+        signal_suffix = "prelim" if dataset_id == PRELIM_DATASET_ID else "ew"
+        nhsn_meta_df = meta_df[(meta_df["data_source"] == "nhsn") & (meta_df["signal"].str.endswith(signal_suffix))]
+        est = timezone(timedelta(hours=-5))
+        last_updated = datetime.fromtimestamp(nhsn_meta_df["last_update"].min(), tz=est)
+
+        # currently set to run twice a week, RECENTLY_UPDATED_DIFF may need adjusting based on the cadence
+        recently_updated_source = (updated_timestamp - last_updated) > RECENTLY_UPDATED_DIFF
```

Comment:
i dont think this math is quite right...  why wouldnt we want to proceed any time socrata has a newer timestamp than we do?  the form you have here has the potential to delay processing if updates are frequent enough or if `RECENTLY_UPDATED_DIFF` is too large.
```suggestion
        socrata_ts = updated_timestamp
        delphi_ts = last_updated
        recently_updated_source = socrata_ts > delphi_ts
```